src/t8_cmesh/t8_cmesh_vtk_to_t8/t8_cmesh_vtk_polydata.cxx

```diff
@@ -92,49 +93,48 @@ t8_read_poly_ext (const char *filename, vtkSmartPointer < vtkPolyData > grid)
     if (!reader->IsFilePolyData ()) {
       t8_errorf
         ("File-content is not polydata. If it is a vtkUnstructuredGrid use the unstructured Grid reader.");
-      return 0;
+      return read_failure;
     }
-    grid->ShallowCopy (reader->GetOutput ());
-    return 1;
+    grid->ShallowCopy (vtkDataSet::SafeDownCast (reader->GetOutput ()));
+    return read_success;
   }
   else if (strcmp (extension, "g") == 0) {
     vtkNew < vtkBYUReader > reader;
     reader->SetGeometryFileName (filename);
     reader->Update ();
-    grid->ShallowCopy (reader->GetOutput ());
-
-    return 1;
+    grid->ShallowCopy (vtkDataSet::SafeDownCast (reader->GetOutput ()));
+    return read_failure;
   }
   else {
     /* Return NULL if the reader is not used correctly. */
     t8_global_errorf ("Please use .ply, .vtp, .obj, .stl, .vtk or .g file\n");
-    return 0;
+    return read_success;
   }
 }
 
-int
+vtk_read_success_t
 t8_read_poly (const char *filename, vtkDataSet * grid)
```

Comment:
If not used elsewhere make `static`.

Reply:
This function is used in t8_cmesh_vtk_reader.cxx:148